config/Dockerfile

```diff
@@ -1,5 +1,5 @@
 # NOTE: Keep this in sync with .ci-operator.yaml
-FROM registry.ci.openshift.org/openshift/release:golang-1.13
+FROM registry.ci.openshift.org/openshift/boilerplate:image-v0.5.2
```

Comment:
Is this a good idea? This means we won't get any security fixes if the underlying image is rebuilt

Reply:
By "underlying image" you mean `release:golang-1.13`? That's true. (To editorialize: relying on the moving-target behavior of a tagged image is just as bad as the producers of that image relying on us relying on that.)

If security (or other) fixes went into the base image, we wouldn't have picked them up anyway until the next time we cut a new version, because the [push script only uploads a new tagged version if we're on a tagged commit](https://github.com/openshift/boilerplate/blob/58c86db5160b06c2d0f3ee7e34f7ccc9ba582b2c/config/push.sh#L34-L38).

Which means deliberate action is required in order to pull in such fixes. Before this PR, a null image rev bump with a new tag. After this PR... something else, probably involving mucking with these files.

But even then, consuming repositories would have to merge a(ny) PR to pick up the changes. (Worth noting that the consumers only use the image for testing/linting/building the operator anyway, not for running it.)

That push script is kind of a mess anyway. It's building and pushing the image with `latest` and `$commit_hash` tags for every PR, even though those are a) duplicates of the most recent versioned image, and b) never used. We could fix it to
- stop doing that
- always rebuild the prior image(s), based on the renamed Dockerfile(s)
- always rebuild the most recent versioned image, which would then be based on ^

The point of this PR is to *not* always rebuild from scratch in prow, and I would really like to preserve that behavior. Having longer postsubmit build times is acceptable though. If we only modify the push script as described above, we achieve that -- but at the cost of the presubmit and postsubmit potentially building slightly different things. However, the only thing *this* repo runs against that image is shell code, so it's "never" going to notice the difference.

Thoughts?

Reply:
Okay, I did this.